Rubberduck.Deployment.Build/RubberduckPostBuildTask.cs

```diff
@@ -0,0 +1,365 @@
+            SourceTlb32 = Path.Combine(sourceDir, Tlb32File);
+            TargetTlb32 = Path.Combine(targetDir, Tlb32File);
+            SourceTlb64 = Path.Combine(sourceDir, Tlb64File);
+            TargetTlb64 = Path.Combine(targetDir, Tlb64File);
+            DllXml = TargetDll + ".xml";
+            TlbXml = TargetTlb32 + ".xml";
+        }
+    }
+
+    public class RubberduckPostBuildTask : AppDomainIsolatedTask
+    {
+        [Required]
+        public string Config { get; set; }
+
+        [Required]
+        public string NetToolsDir { get; set; }
+
+        [Required]
+        public string WixToolsDir { get; set; }
+
+        [Required]
+        public string SourceDir { get; set; }
+
+        [Required]
+        public string TargetDir { get; set; }
+
+        [Required]
+        public string ProjectDir { get; set; }
+
+        [Required]
+        public string IncludeDir { get; set; }
+
+        [Required]
+        public string FilesToExtract { get; set; }
+
+        private string RegFilePath =>
+            Path.Combine(Path.Combine(ProjectDir, "LocalRegistryEntries"), "DebugRegistryEntries.reg");
+
+        private string _rootPath;
+        private string _batchPath;
+
+        public override bool Execute()
+        {
+            var result = true;
+            try
+            {
+                this.LogMessage(FormatParameterList());
+
+                CleanOldImports(ProjectDir);
+
+                var dllFiles = FilesToExtract.Split(new[] {"|"}, StringSplitOptions.RemoveEmptyEntries);
+
+                var message = SetVCToolsPath()
+                    ? "No C++ build tools found; using tlbexp.exe to generate TLBs."
+                    : "C++ build tools found; using midl.exe to generate TLBs.";
+                this.LogMessage(message);
+                
+                foreach (var dllFile in dllFiles)
+                {
+                    ProcessDll(dllFile);
+                }
+
+                if (Config == "Debug")
+                {
+                    UpdateAddInRegistration();
+                }
+            }
+            catch (Exception ex)
+            {
+                this.LogError(ex);
+                result = false;
+            }
+
+            return result;
+        }
+
+        private string FormatParameterList()
+        {
+            var sb = new StringBuilder();
+            sb.AppendLine("Parameters provided:");
+            sb.AppendLine($"          Config: {Config}");
+            sb.AppendLine($"     NetToolsDir: {NetToolsDir}");
+            sb.AppendLine($"     WixToolsDir: {WixToolsDir}");
+            sb.AppendLine($"       SourceDir: {SourceDir}");
+            sb.AppendLine($"       TargetDir: {TargetDir}");
+            sb.AppendLine($"      ProjectDir: {ProjectDir}");
+            sb.AppendLine($"     IncludesDir: {IncludeDir}");
+            sb.AppendLine($"  FilesToExtract: {FilesToExtract}");
+            sb.AppendLine(string.Empty);
```

Comment:
opportunity for fluent API? does AppendLine support chaining?

Reply:
#TIL SB is fluent.